### DISCUSSION TITLE
:MasonInstall pyright - spawn: npm failed with exit code - and signal - . npm is not executable.

### DISCUSSION THREAD

**Discussion by yanniskatsaros** [mason-org/mason.nvim#674] [Q&A] [answered]
Hello!

Relatively new Neovim user here and first-time user of Mason. I'm attempting to install a new language server using `:MasonInstall pyright` but I get the error: `spawn: npm failed with exit code - and signal - . npm is not executable`. I get this error for a few different language servers I've attempted to install (presumably because they rely on `npm`).

I have `npm` installed, but is there something I'm missing here? Appreciate any help!

**Comment by williamboman** [comment, marked as answer]
Hello! Do you by chance use volta to manage Node & npm? There's been [some reports](https://github.com/williamboman/mason.nvim/discussions/423) that it's not compatible. I've yet to investigate what exactly volta does in more detail (I believe they shim executables somehow).

Mason only expects the `node` and `npm` executables to be available in the neovim session's `PATH` - so whatever you use to manage these shouldn't really matter, so I'd really be interested in knowing what volta does differently.

Does the following vim commands produce anything?
```vim
:echo exepath("node")
:echo exepath("npm")
```

**Reply by yanniskatsaros** [reply]
Hey, thanks so much for getting back to me! I tried running those commands and interestingly enough they don't produce any output... I don't use volta, but I do use `nvm` to manage Node and npm versions. From my terminal:

```bash
$ which node
/Users/yannis/.nvm/versions/node/v16.15.1/bin/node

$ which npm
/Users/yannis/.nvm/versions/node/v16.15.1/bin/npm
```

Perhaps this is causing a similar issue?

**Reply by yanniskatsaros** [reply]
I just checked `:echo $PATH` (inside Neovim) and the path is different from when I check `echo $PATH` in my terminal. 🤔 not sure why this is the case - investigating further.

Is it standard practice to manually set/modify the `$PATH` parameter inside of Neovim (in a similar way to how you `export $PATH=...` in `~/.zshrc` for example)?

**Reply by williamboman** [reply]
Do you use a GUI frontend for Neovim? If you launch it inside the same terminal session the Neovim instance should inherit the shell's environment.

> Is it standard practice to manually set/modify the `$PATH` parameter inside of Neovim (in a similar way to how you `export $PATH=...` in `~/.zshrc` for example)?

Not sure there's such thing as a standard practice. If you'd like Neovim to run with a specific `PATH` then sure it makes sense to modify it. I personally let my shell manage the environment.

**Reply by yanniskatsaros** [reply]
That's an excellent point about the GUI frontend, yes I use Neovide. I didn't even think of that. I tried running `nvim` directly from my terminal and my `$PATH` does indeed inherit the shell's environment. Thank you for your help on this I really appreciate it - I apologize that the issue didn't end up being related to `mason`.

**Comment by godsonsakawa** [comment]
it works :)

**Comment by rovellipaolo** [comment]
I had the same issue while running the flatpak version of Neovim (see; https://github.com/neovim/neovim/wiki/Installing-Neovim#flatpak).
To make it work I had to install the `node` SDK extension and launch Neovim with it enabled:
```
$ flatpak install flathub org.freedesktop.Sdk.Extension.node18 --user
$ FLATPAK_ENABLE_SDK_EXT=node18 flatpak run io.neovim.nvim
```

**Comment by Goralive** [comment]
I also faced the issue. I was using fvm(nvm just re written on Rust) and forgot to use version. So after I select version to use and restart my terminal the version was inherit from the shell's environment and successfully applied.
